cyclic_boosting/base.py

```diff
@@ -338,11 +339,21 @@ def _check_weights(self):
         if self.weights is None:
             raise RuntimeError("The weights have to be initialized.")
 
-    def _init_features(self):
+    def _init_features(self, X=None):
         """
         Initializes the ``features`` and binds the data belonging to a feature
         to it.
         """
+        if X is not None:
+            X_dtypes = X.dtypes.to_dict()
+            for key in X_dtypes:
+                if np.issubdtype(X_dtypes[key], np.integer):
+                    if self.feature_properties[key] is None:
+                        self.feature_properties[key] == flags.IS_UNORDERED
+                if np.issubdtype(X_dtypes[key], np.float):
+                    if self.feature_properties[key] is None:
+                        self.feature_properties[key] == flags.IS_CONTINUOUS
+
```

Comment:
This functionality should be here: https://github.com/Blue-Yonder-OSS/cyclic-boosting/blob/490df233c6acfe7b870228184cf4cf5b887f30f3/cyclic_boosting/features.py#L270